Disabled CSRF protection for IPN view

anvio opened the pull request:
As of Django version 1.2 the CSRF protection is enabled by default. To make the IPN view work I had to disable the CSRF protection with the @csrf_exempt decorator .

Commit:
Split bulk of ipn view into helper, enable tracking of views

Commit:
Revert "+ merged signals from ipn/pdt"

This reverts commit 6039d079071b74150ffbbcc969e39bbbc30f9fd7.

	Partially.

Conflicts:
	(apparently none)

Some tweaking was required, this is a partial revert,
to restore existance of:
	new file:   standard/ipn/signals.py

Commit:
Move newly-restored signals file to proper place

Commit:
Most important patch for fixing exchange rates in IPN posts

PayPal uses an unspecified amount of decimals when returning
an IPN with an exchange rate - this takes that into account!

ALTER TABLE paypal_ipn ALTER COLUMN exchange_rate TYPE numeric(64, 16);

Commit:
Fix object referenced before assignment bug

If the PayPal POST doesn't pass validation, we can't set
flags to it without initializing.

This still has the problem that the ipn object becomes
empty, it should be populated with real data.

Commit:
Verify from_view when looking for duplicate in model method

Commit:
Verify from_view when looking for duplicate in model method
(cherry picked from commit 0e1f2c0e3517b225be9abf27c7c139136a8822b7)

Commit:
Make model aware of multiple states

Commit:
When looking for duplicates, always take current status into account

mthornhill committed:
resolve send_signals conflict

mthornhill committed:
Merge branch 'multistate' of git://github.com/mjtorn/django-paypal

mthornhill committed:
merged all django-paypal branches and fixed test cases

mthornhill committed:
use separate functions for rendering sandbox and real payment buttons

mthornhill committed:
add a test to ensure that the cm field gets saved in the custom field of PDT objects

mthornhill committed:
use the SANDBOX_ENDPOINT for pdt when settings.DEBUG=True

mthornhill committed:
merging with john boxalls master branch

mthornhill committed:
merged changes from john boxalls master branch

mthornhill committed:
initial south migrations to allow for database changes in the future

mthornhill committed:
used --initial when creating startmigration

mthornhill committed:
merges from git://github.com/johnboxall/django-paypal.git master up to http://github.com/johnboxall/django-paypal/commit/0c8aeec1c319a08ce1bfdf828534d01b69b8fa27

Commit:
doDirectPayment and createRecurringPaymentsProfile now return an NVP object rather than a boolean. Added support for manangeRecurringPaymentsProfileStatus.

Commit:
Added two new signals: recurring_create and recurring_payment. Some notes about possible signal changes.

Commit:
Updated signal list.

Commit:
Added 3 more recurring payment signals.

Commit:
Add in check for value

Commit:
Changed package layout. Added setuptools script. Added CONTRIBUTORS.

Commit:
API ref link, as its extremely useful.

Commit:
Moved payment_was_successful signal into API operations. Added support for GetExpressCheckoutDetails

Commit:
Added support for UpdateRecurringPaymentsProfile

Commit:
Slip on copy/paste docs. Added support for GetTransactionDetails

Commit:
Force arguments to be strings

Commit:
Credit card expiry widget can now accept the formatted string value.

Commit:
just some whitespace cleanup

Commit:
Added support for recurring_payment_profile_cancel IPN callback. Tweaked IPN view to handle N/A values on datetimes.

Commit:
Changed all flag boolean checking in Pro to use exception handling with the new PayPalFailure exception.

Commit:
Added paypal.__version__

Commit:
missing import in setup.py

Commit:
updated for pypi

Commit:
updated classifiers

Commit:
Support for django-idmapper

Commit:
Fix for failing unit test

danielgtaylor committed:
Allow duplicate transaction ids for payments which go from pending to completed. When this happens the payment should not be flagged but processed like any other successful payment. This can happen for large amounts of money, currency conversions, transactions from certain countries, and other reasons at the discretion of Paypal.

mthornhill committed:
merge from HEAD of dcramers fork

mthornhill committed:
duplicate_txn_id only takes one argument

mthornhill committed:
why not?

Commit:
Increment version to 0.1.2

Commit:
Increment version to 0.1.2

Commit:
payment_was_succes[s]ful

Commit:
payment_was_succes[s]ful

Commit:
Added support for PAYPAL_DEBUG rather than relying on DEBUG

Commit:
Added support for PAYPAL_DEBUG rather than relying on DEBUG

Commit:
#10 add db_index=True to txn_id

Commit:
#10 add db_index=True to txn_id

mthornhill committed:
include template html files in build

Commit:
Merge branch 'master' of git://github.com/mthornhill/django-paypal

Commit:
Somehow the signals were moved out of the standard path

Commit:
Looking into it a bit more, standard.signals should not even have existed

mikexstudios committed:
Added csrf_exempt tag for ipn callback view. Minor comments in code.

dcramer committed:
Improve admin panel for NVPs

dcramer committed:
Added standalone unit test runner. Fixed test_doDirectPayment_invalid logic.

dcramer committed:
Merge branch 'master' of github.com:dcramer/django-paypal

dcramer committed:
Include IPN tests in runner

dcramer committed:
Deal with credit card numbers that include - and spaces

wolever committed:
Adding instructions for getting PayPal API credentials.

wolever committed:
Adding note about test credit cards in PP Pro sandbox.

dcramer committed:
Clean up docs a bit

dcramer committed:
Reorder COUNTRIES

chrischambers committed:
Adding payment_profile_created signal

chrischambers committed:
Stripping port-info from PayPalNVP ipaddress field

This is necessary to avoid errors from postgres backends.

john-kurkowski committed:
Add Donation button to standard form button types. Expose possible button types on the form class.

dcramer committed:
Merge branch 'master' of http://github.com/chrischambers/django-paypal into chrischambers-master

kevin1024 committed:
fixed bug where WPP would send the payment success signal twice (The signal is also sent by the model.)

kevin1024 committed:
updated README file to remove dead link and add missing step to WPP instructions

kevin1024 committed:
fix formatting for WPP template

rozza committed:
All image configs can now be overridden

mpdaugherty commented:
I also had to make this change and was about to submit my own pull request when I saw there is one already open.  Seems to be a common problem: http://stackoverflow.com/questions/2580375/ipn-delivery-failed-http-error-code-403-forbidden

Therefore, for what it's worth, I support this request.

anvio commented:
You may want to have a look at the fork of dcramer 
https://github.com/dcramer/django-paypal
It is much more up-to-date.

mpdaugherty commented:
Thanks, Anvio

anvio committed:
internationalization fix

Commit:
Fixed initialization of PayPalSharedSecretEncryptedPaymentsForm

dcramer committed:
Merge pull request #9 from zbyte64/master

Fixes initialization of PayPalSharedSecretEncryptedPaymentsForm

anvio committed:
Merge branch 'master' of git://github.com/dcramer/django-paypal

anvio committed:
when subscriptions failed the retry_at datetime field was not parsed properly so the IPN was ignored, added new signal for failed subscriptions

anvio committed:
disabled some unit tests as the test suite is broken in django 1.3.1

oliverzheng committed:
Making standard.ipn tests pass.

On MySQL, ordering by date does not give the desired ordering of getting the
most recently inserted row first. We need to order by the primary key too.

dcramer committed:
Merge pull request #14 from MTsoul/master

Making tests pass for standard.ipn for MySQL.

jasonculverhouse committed:
add signals and tests for recurring_skipped, recurring_failed ipn

jasonculverhouse committed:
change file permissions

jasonculverhouse committed:
make the recurring_payment test into a negative test case, this signal is never called by the code because this ipn has a txn_id

jasonculverhouse committed:
make tests work in isolation by removing and replacing all signals

dcramer committed:
Merge pull request #15 from jasonculverhouse/master

Added signals for recurring_payment_skipped and recurring_payment_failed

jasonculverhouse committed:
remove redundant use of recurring_failed

dcramer committed:
Merge pull request #17 from jasonculverhouse/master

Fixed use of recurring_failed in the ipn_tests

spookylukey committed:
Improved IPN tests to post data the way PayPal does

spookylukey committed:
Handle saving of query the way PayPal needs

spookylukey committed:
Fixed encoding problem for parameters sent in PayPal IPN POST data

spookylukey committed:
Corrected comments on some fields on standard model

atul-bhouraskar committed:
Handle Paypal missing payment status types.

The following payment status types are being added:
- Canceled_Reversal
- Refunded
- Reversed
- Voided

The payment_status field in PayPalStandardBase is being
modified. The max_length is now 17 to handle new types.
Corresponding migrations for ipn and pdt exist.

The following new IPN signals are being added:
- payment_was_refunded
- payment_was_reversed

The following new methods are being added to PayPalStandardBase:
- is_refund
- is_reversed

Added test for the new signals.

atul-bhouraskar committed:
Ensure that _verify_postback passes through any flags set as the result of the postback verification to the PayPalPDTForm.

When a PDT trnsaction is flagged, the flags set (if any) are not passed to the form that is used to update the pdt object.
This commit fixes this issue.

Commit:
merged commits from atul-bhouraskar

Commit:
merged repo from spookylukey

Commit:
fixed immutable QueryDict bug

Commit:
fixing "This QueryDict instance is immutable"

Commit:
made payer email searchable in admin
